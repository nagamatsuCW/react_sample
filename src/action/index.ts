export const actionTypes = {
  INCREMENT: "INCREMENT",
  DECREMENT: "DECREMENT"
}


export const increment = () => {
  return {
    type: actionTypes.INCREMENT
  } as const
}

export const decrement = () => {
  return {
    type: actionTypes.DECREMENT
  } as const
}

export const actions = { increment, decrement }

export type ActionsType = ReturnType<typeof increment | typeof decrement>