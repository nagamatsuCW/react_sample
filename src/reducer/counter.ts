import { ActionsType } from '../action'
export interface CounterState {
  count: number
}

const initialState: CounterState = {
  count: 0
};

const counter = (state:CounterState = initialState, action:ActionsType) => {
  switch (action.type) {
    case 'INCREMENT':

      return {
        ...state,
        count: state.count + 1
      }
    case 'DECREMENT':
      return {
        ...state,
        count: state.count - 1
      }
    default:
      return state
  }
}

export default counter