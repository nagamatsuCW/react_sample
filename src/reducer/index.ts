import { defaultCipherList } from 'constants'
import { combineReducers } from 'redux'
import counter from './counter'
import { connectRouter } from 'connected-react-router'
import history, { History } from '../history'


const createRootReducer = (history: History) => combineReducers({
  router: connectRouter(history),
  counter
})
export default createRootReducer

export type RootState = ReturnType<typeof createRootReducer>