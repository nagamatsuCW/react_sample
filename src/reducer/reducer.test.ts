import reducer, { initialCounterState as initialState } from './counter'
import { decrement, increment } from '../action'


describe('reducer test', () => {
  it('increment', () => {
    const next = reducer(initialState, increment())
    expect(next).toMatchSnapshot()
  })
  it('decrement', () => {
    const next = reducer(initialState, decrement())
    expect(next).toMatchSnapshot()
  })
})