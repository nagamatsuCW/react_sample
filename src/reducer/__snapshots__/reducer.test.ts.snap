// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer test decrement 1`] = `
Object {
  "count": -1,
}
`;

exports[`reducer test increment 1`] = `
Object {
  "count": 1,
}
`;
